akka-stream-tests/src/test/scala/akka/stream/scaladsl/One2OneBidiFlowSpec.scala

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import org.scalactic.ConversionCheckedTripleEquals
+import akka.stream.ActorMaterializer
+import akka.stream.testkit.AkkaSpec
+import akka.stream.testkit._
+
+class One2OneBidiFlowSpec extends AkkaSpec with ConversionCheckedTripleEquals {
+  import FlowGraph.Implicits._
+
+  implicit val mat = ActorMaterializer()
+
+  /**
+   * Wraps a flow with one-to-one guarding, which
+   *
+   * 1. triggers an `OutputTruncationException` if the wrapped stream terminates
+   *    before exactly one output has been produced for each input element
+   * 2. Backpressures surplus output elements (i.e. ones that are being produced without
```

Comment:
Doesn't that contradict 3.? Or otherwise, if it is allowed to produce several output elements per input element, how do you associate the surplus with incoming elements?